src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java

```diff
@@ -544,31 +565,93 @@ private Boolean isPassable(final BlockState blockState)
         return blockState.getMaterial() == Material.LEAVES;
     }
 
+
     /**
      * Check if distance to block changed and if we are not moving for too long, try to get unstuck.
      *
      * @return false
      */
-    private boolean checkIfStuckOnLeaves()
+    private boolean checkIfStuck()
     {
         if (!worker.getNavigator().noPath())
         {
-            Path pathToTree = worker.getNavigator().getPath();
-            if (pathToTree != null && pathToTree.getCurrentPathLength() > pathToTree.getCurrentPathIndex())
-            {
-                PathPoint next = pathToTree.getPathPointFromIndex(pathToTree.getCurrentPathIndex());
-                BlockPos nextPos = new BlockPos(next.x, next.y, next.z);
-                BlockPos nextPosUp = new BlockPos(next.x, next.y + 1, next.z);
-
-                if (world.getBlockState(nextPos).getBlock().isIn(BlockTags.LEAVES))
-                {
-                    mineBlock(nextPos);
+            Path path = worker.getNavigator().getPath();
+            if(path != null) {
+                if (path.getCurrentPathLength() > path.getCurrentPathIndex()) {
+                    return true;
                 }
-                else if (world.getBlockState(nextPosUp).getBlock().isIn(BlockTags.LEAVES))
-                {
-                    mineBlock(nextPosUp);
+                if(path.getCurrentPathLength() == 0){
+                    return true;
                 }
             }
+            else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Mines leaves on path and above.
+     */
+    private void tryUnstuck(){
+
+        if (!worker.getNavigator().noPath()) {
+            Path path = worker.getNavigator().getPath();
+            if (path != null) {
+                // Unstuck with path
+                ArrayList<BlockPos> checkPositions = new ArrayList<>();
+                PathPoint next = path.getPathPointFromIndex(path.getCurrentPathIndex());
+                // Blocks in front of the worker
+                checkPositions.add(new BlockPos(next.x, next.y, next.z));
+                checkPositions.add(new BlockPos(next.x, next.y + 1, next.z));
+                // Block above the worker
+                checkPositions.add(new BlockPos(worker.getCurrentPosition().getX(), worker.getCurrentPosition().getY() + 2, worker.getCurrentPosition().getZ()));
+
+                mineIfEqualsBlockTag(checkPositions, BlockTags.LEAVES);
+                return;
+            }
+        }
+
+        // General unstuck
+        ArrayList<BlockPos> checkPositions = new ArrayList<>();
```

Comment:
we are already using the pathing to see which leaves need to be removed, I don't think we need this general remove all around worker